### PR TITLE
feat: add domain_to_ascii using icu

### PR DESCRIPTION
**Pull request by anonrig**

### CODE REVIEW

**Comment by lemire** [comment]
@anonrig Unless you object, once I am done reviewing, I will merge unless you tell me not to.

**Comment by anonrig** [comment]
I’m good, we can merge @lemire

File: README.md [hunk 1]
```diff
@@ -4,10 +4,14 @@ Ada is a fast and spec-compliant URL parser written in C++.
...
 
+## Requirements
+
+- [ICU](https://icu.unicode.org)
+
 ## Usage (CMake)
 
 ```
-cmake -B build
+cmake -DCMAKE_PREFIX_PATH=/opt/homebrew/opt/icu4c -B build
```
Comment (lemire) [inline README.md:14]: That should not be necessary in general. I will revisit that.

**Comment by lemire** [comment]
Ok. Let us merge this. It is not perfect but it is quite good enough.

**Pull request merged**